src/app/WriteHandler.cpp

```diff
@@ -74,6 +74,8 @@ CHIP_ERROR WriteHandler::OnWriteRequest(Messaging::ExchangeContext * apExchangeC
 
 exit:
     ChipLogFunctError(err);
+    // Keep Shutdown() from double-closing our exchange.
+    mpExchangeCtx = nullptr;
```

Comment:
Same question as above, where the exchange is closed after ProcessWriteRequest, shall we call mpExchangeCtx->WillSendMessage(); in ProcessWriteRequest as ProcessReadRequest?

Reply:
ProcessWriteRequest does not do any async work right now.   We send the response immediately after calling `ProcessWriteRequest` and want to close the exchange right after that.

We _could_ call `WillSendMessage` in `ProcessWriteResponse`.  That flag would just get cleared by the send that `SendWriteResponse` does.  Unless that function fails to reach the send call, in which case we'd need to close manually.  Doable, but a bit more complicated than what we have now.  If we start doing async work in WriteHandler, we should add the `WillSendMessage` at that point.